### ISSUE TITLE
[ON HOLD Payment 4/3/24] [Violations] [$500] Don't show RBR on the LHN for transaction threads with violations if the report has already been reimbursed/settled

### ISSUE THREAD

**Issue by cead22** [Expensify/App#34548]
- [Slack discussion](https://expensify.slack.com/archives/C02MW39LT9N/p1704465959260069?thread_ts=1704464705.512839&cid=C02MW39LT9N) (internal)
- Transaction threads for transactions that have violations show up on the LHN with a RBR for the use that's requesting money
- The idea behind RBR is to show the user a path to take an action to fix a potential issue
- This is reasonable for the user requesting money because their requests/transactions may contain violations that can prevent them from getting approved and paid
- Once the requests are paid, we agreed to continue showing violations in order to show that the request may not be in compliance with the current workspace rules, which is useful for the users requesting money, and also users paying those requests
- However, the RBR specifically on the LHN isn't as relevant because the main idea behind that is to show a path to fix an issue, but in this case, the "issue" is the request has violations, but since they've been paid/settled, this isn't important for the requestor to do

Given that, let's not show RBR on the LHN for transaction threads with violations, if the transactions/money requests have been paid/settled

@trevor-coleman @lindboe can you please comment here so I can assign you? Thanks

---

If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.4.43-14
**Reproducible in staging?:** Presumably
**Reproducible in production?:** Presumably
**Issue reported by:** @situchan https://github.com/Expensify/App/pull/35449#issuecomment-1961141823

## Action Performed:
- Make sure you're on the violations beta
- Request money without adding a category from workspace that requires category (it should have a missing category violation)
- Submit the request if you need to
- Approve the request as a workspace admin
- Open the transaction thread as the requester. You may need to navigate to another chat, and then back to the transaction thread

## Expected Result:
- RBR should **not** be shown on the LHN for the submitter

## Actual Result:
- RBR is shown on the LHN for the submitter

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence
<img width="1703" alt="image" src="https://github.com/Expensify/App/assets/165133/54a415cd-2bab-484f-9a6c-6a8e43aaafd7">

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01e69e67564d338f8f</li>
        <li>Upwork Job ID: 1761101186954891264</li>
        <li>Last Price Increase: 2024-02-23</li>
<li>Automatic offers: </li>
<ul>
<li>situchan | Reviewer | 0</li>
<li>FitseTLT | Contributor | 0</li>
</ul></li>
    </ul>
</details>

**Comment by trevor-coleman** [comment]
Commenting for assignment

**Comment by cead22** [comment]
ReportUtils PR was just merged, so we're starting to work on other issues, but that one was priority

**Comment by cead22** [comment]
Still a priority but other issues are being worked on first

**Comment by lindboe** [comment]
@cdanwards this is the ticket you were pulling into optimistic IDs, right? since we're already making RBR changes there?

**Comment by cead22** [comment]
WIP

**Comment by situchan** [comment]
Happy to take this as C+ since I already have context

**Comment by cead22** [comment]
- @situchan can you check the reproduction steps and make any relevant updates?
- @CortneyOfstad let's manage this like a regular bug and get proposals, and @situchan will be the C+ reviewing

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Don't show RBR on the LHN for transaction threads with violations if the report has already been reimbursed/settled
### What is the root cause of that problem?
We are not checking if it is setteled here
https://github.com/Expensify/App/blob/d701972794f6c93d9df4848b5ddc42d99f00946f/src/libs/SidebarUtils.ts#L238-L239
### What changes do you think we should make in order to solve the problem?

Change it to 
```
    result.brickRoadIndicator = (hasErrors || hasViolations) && !ReportUtils.isSettled(report) ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : '';

```
### What alternative solutions did you explore? (Optional)

**Comment by situchan** [comment]
Repro step looks correct.

**Comment by situchan** [comment]
> @situchan , how do we enable violations beta on our expensify account ?

You can raise request on slack

**Comment by CortneyOfstad** [comment]
@FitseTLT bump on the comment [here](https://github.com/Expensify/App/issues/34548#issuecomment-1961841624)

@codinggeek2023 i didn't see any requests in Slack for enabling violations, but if you need any assistance with that, just let me know! Thanks!

**Comment by FitseTLT** [comment]
Oh thx @CortneyOfstad 

@situchan I have received the permission and verified my solution works. U can proceed 👍

**Comment by situchan** [comment]
@FitseTLT's [proposal](https://github.com/Expensify/App/issues/34548#issuecomment-1961837520) looks good to me.
🎀 👀 🎀 C+ reviewed

**Comment by cead22** [comment]
@FitseTLT I think the change you shared under `Change it to` is better than putting it inside `ReportUtils.doesTransactionThreadHaveViolations` since it's used in other places, or in the line with  `result.brickRoadIndicator`

**Comment by CortneyOfstad** [comment]
@FitseTLT can you provide an update on when you think you'll have a PR created? EOD, EOW? Thanks!

**Comment by CortneyOfstad** [comment]
@zanyrenney I am heading OoO, but will be back March 11th. We're waiting on @FitseTLT to provide an update on when the PR will be created at this stage. Thanks!

**Comment by zanyrenney** [comment]
nice, thank you @FitseTLT

**Comment by FitseTLT** [comment]
PR is on review stage 👍

**Comment by JmillsExpensify** [comment]
Good movement on the linked PR. Ideally we can even get it merged before the week closes out.

**Comment by cead22** [comment]
PR is merged

**Comment by sakluger** [comment]
FYI this still seems to be broken for me. I left a more detailed comment in the PR: https://github.com/Expensify/App/pull/37767#issuecomment-2015654957.

**Comment by cead22** [comment]
I'm pretty sure that error is unrelated, and this is happening  because OpenReport is failing with a JSON error
<img width="1670" alt="image" src="https://github.com/Expensify/App/assets/165133/75e32ccb-ad0d-4146-9c3c-b18cbc84d77e">

It looks like this started on Friday 
<img width="1672" alt="image" src="https://github.com/Expensify/App/assets/165133/f786c3dc-f13a-423b-83bb-d881844f678e">

**Comment by cead22** [comment]
I found the issue and submitted a PR

**Comment by cead22** [comment]
That said, the problem is unrelated to this GH issue, so I'm going to close this one since it's done

**Comment by FitseTLT** [comment]
@cead22 What about Payments?? 😆

**Comment by cead22** [comment]
Argh, they always get me. I see the upwork jobs were created, but not sure what else we need for payments -- @CortneyOfstad @zanyrenney can you help with this please?

**Comment by CortneyOfstad** [comment]
No worries — since it was just deployed today, we will need to wait the 7-day period, so going to adjust the title to reflect the payment date if there are no regressions 👍

**Comment by CortneyOfstad** [comment]
Payment has been completed! Will post payment summary below.

@situchan does this need a regression test? For some reason the checklist is not populating, so just wanted to confirm. Thanks!

**Comment by CortneyOfstad** [comment]
## Payment Summary

@FitseTLT (Contributor) — paid $500 via Upwork
@situchan (C+) — paid $500 via Upwork
